### PR TITLE
Add config_groups support

### PR DESCRIPTION
**Pull request by bagasse**
Add config groups support.
Needed by #366 to add config_groups tests on hostgroups.

### CODE REVIEW

**Comment by bagasse** [comment]
Sanity error related to #369

File: plugins/modules/foreman_config_group.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+from ansible.module_utils.foreman_helper import (
+    split_fqn,
+    ForemanEntityApypieAnsibleModule,
+    sanitize_entity_dict,
+)
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'puppetclasses': 'puppetclass_ids'
```
Comment (mdellweg) [inline plugins/modules/foreman_config_group.py]:
```suggestion
    'puppetclasses': 'puppetclass_ids',
```

**Comment by bagasse** [comment]
@mdellweg, thanks for review. Suggestion applied (+ some variables names fixes)

**Comment by bagasse** [comment]
Rebased to fix sanity checks.

File: tests/test_playbooks/config_group.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        config_group_state: present
+        expected_change: true
+    - include: tasks/config_group.yml
+      vars:
+        config_group_state: present
+        config_group_puppetclasses: []
+        expected_change: false
+    - include: tasks/config_group.yml
+      vars:
+        config_group_puppetclasses: ['ntp', 'ntp::config']
```
Comment (evgeni) [inline tests/test_playbooks/config_group.yml]: does that mean we need to import the `ntp` puppet class before this playbook can run?

Comment (evgeni) [reply tests/test_playbooks/config_group.yml]: (or be recorded, to be precise)

Comment (bagasse) [reply tests/test_playbooks/config_group.yml]: Yes, as stated in the comment at the top of the file.

File: plugins/module_utils/foreman_helper.py [hunk 3]
```diff
@@ -393,12 +393,12 @@ def parse_template_from_file(file_name, module):
 
 
 # Helper for titles
-def split_fqn(title):
+def split_fqn(title, separator='/'):
     """ Split fully qualified name (title) in name and parent title """
-    fqn = title.split('/')
+    fqn = title.split(separator)
```
Comment (evgeni) [inline plugins/module_utils/foreman_helper.py]:
```suggestion
    fqn = title.split(separator, 1)
```

then you don't need to `join` anymore :)

File: plugins/modules/foreman_config_group.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    entity = module.find_resource_by_name('config_groups', name=entity_dict['name'], failsafe=True)
+
+    if not module.desired_absent:
+        if 'puppetclasses' in entity_dict:
+            # puppet classes API return puppet classes grouped by puppet module name
+            puppet_classes = []
+            for puppet_class in entity_dict['puppetclasses']:
+                search = 'name="{}"'.format(puppet_class)
+                module_name, class_name = split_fqn(puppet_class, '::')
```
Comment (evgeni) [inline plugins/modules/foreman_config_group.py]: where are those two used?

File: plugins/modules/foreman_config_group.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                results = module.list_resource('puppetclasses', search)
+                if len(results) == 1 and len(results[results.keys()[0]]) == 1:
+                    result = results[results.keys()[0]]
```
Comment (evgeni) [inline plugins/modules/foreman_config_group.py]: this seems unsed too

File: plugins/modules/foreman_config_group.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+    if not module.desired_absent:
+        if 'puppetclasses' in entity_dict:
+            # puppet classes API return puppet classes grouped by puppet module name
+            puppet_classes = []
+            for puppet_class in entity_dict['puppetclasses']:
+                search = 'name="{}"'.format(puppet_class)
+                module_name, class_name = split_fqn(puppet_class, '::')
+                results = module.list_resource('puppetclasses', search)
+                if len(results) == 1 and len(results[results.keys()[0]]) == 1:
```
Comment (evgeni) [inline plugins/modules/foreman_config_group.py]:
```suggestion
                if len(results) == 1 and len(results.values()[0]) == 1:
```
IMHO better readable (you want to ensure the first (and only) entry of the dict has len 1.

File: plugins/modules/foreman_config_group.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+                    result = results[results.keys()[0]]
+                    puppet_classes.append({'id': results[results.keys()[0]][0]['id']})
```
Comment (evgeni) [inline plugins/modules/foreman_config_group.py]: this is hard to read.

File: tests/test_playbooks/config_group.yml (hunk 2 above)
Comment (evgeni) [reply tests/test_playbooks/config_group.yml]: Right. Comments. Sorry :)

File: plugins/module_utils/foreman_helper.py (hunk 3 above)
Comment (bagasse) [reply plugins/module_utils/foreman_helper.py]: Thanks for the pointer :), but I reverted this change as I don't use split_fqn() anymore.

File: plugins/modules/foreman_config_group.py (hunk 4 above)
Comment (bagasse) [reply plugins/modules/foreman_config_group.py]: Not used anymore.

File: plugins/modules/foreman_config_group.py (hunk 5 above)
Comment (bagasse) [reply plugins/modules/foreman_config_group.py]: Yes, sorry for the noise...

File: plugins/modules/foreman_config_group.py (hunk 6 above)
Comment (bagasse) [reply plugins/modules/foreman_config_group.py]: Applied, thanks !

File: plugins/modules/foreman_config_group.py (hunk 7 above)
Comment (bagasse) [reply plugins/modules/foreman_config_group.py]: Added some comment and simplified with your suggestion above.

**Comment by bagasse** [comment]
@evgeni, thanks a lot for the review ! I think that I fixed all that you pointed, please take another look on the part that is hard to read if it's ok for you now :)

**Comment by evgeni** [comment]
Thanks @bagasse!

**Pull request merged**